### PR TITLE
Merge main into int8-sweeps

### PR DESCRIPTION
**Pull request by gobbleturk**

### CODE REVIEW

File: MaxText/configs/base.yml [hunk 1]
```diff
@@ -63,12 +63,12 @@ aqt_rng_type: "jax.uniform"
 
 # Global parameter scale needs to be a power of 2. If you want finer grained control of the model sizes
 # then you should explicitly set base_embed_dim, base_num_heads, base_mlp_dim, base_num_decoder_layers and/or head_dim.
-global_parameter_scale: 0 # has to be set explicitely.
```
Comment (lukaszlew) [inline MaxText/configs/base.yml:66]: Let's keep 0

File: MaxText/configs/base.yml [hunk 2]
```diff
@@ -121,13 +122,14 @@ num_slice: 1 # used by run-sweep.py . will be copied to the other yml file.
...
-vocab_relative_path: "vocabs" # Assumes we're allowed
-dataset_name: "c4/en:3.0.1"
-eval_dataset_name: "c4/en:3.0.1"
-eval_split: "validation"
-per_device_batch_size: 4
+vocab_relative_path: "vocabs"  # Assumes we're allowed
+dataset_name: 'c4/en:3.0.1'
+eval_dataset_name: 'c4/en:3.0.1'
+eval_split: 'validation'
+per_device_batch_size: 12.0
```
Comment (lukaszlew) [inline MaxText/configs/base.yml:129]: Let's set it to -1 to force setting explicitely

File: MaxText/configs/base.yml [hunk 3]
```diff
@@ -140,6 +142,7 @@ clip_by_block_rms: 0.0 # no clipping default
 clip_by_ucb: 0.0 # no clipping default , exact value is ignored if above 0.0
 
 # Maximum length cutoff for training examples.
+max_target_length: 2048
```
[the remark this answers is not in the document]
Comment (lukaszlew) [reply MaxText/configs/base.yml:145]: Remove 1024

File: MaxText/input_pipeline.py [hunk 4]
```diff
@@ -216,11 +217,12 @@ def preprocess_dataset(config: ml_collections.ConfigDict,
       tokenizer.TokenizeOp(sp_tokenizer), num_parallel_calls=AUTOTUNE)
 
   # Set global batch size.
-  batch_size = config.per_device_batch_size * global_mesh.size
+  global_batch_size_to_load = config.global_batch_size_to_load
```
Comment (lukaszlew) [inline MaxText/input_pipeline.py:220]: I didn't see global_batch_size_to_load in yml, what's up?

**Review by lukaszlew — APPROVED** [review]

File: MaxText/input_pipeline.py (hunk 4 above)
Comment (gobbleturk) [reply MaxText/input_pipeline.py:220]: global_batch_size_to_load is defined in [pyconfig.py](https://github.com/google/maxtext/blob/main/MaxText/pyconfig.py#L114)

File: MaxText/configs/base.yml [hunk 5]
```diff
@@ -63,12 +63,12 @@ aqt_rng_type: "jax.uniform"
 
 # Global parameter scale needs to be a power of 2. If you want finer grained control of the model sizes
```
[the remark this answers is not in the document]
Comment (lukaszlew) [reply MaxText/configs/base.yml:64]: I'll do that.

File: MaxText/configs/base.yml (hunk 1 above)
Comment (gobbleturk) [reply MaxText/configs/base.yml:66]: Done.

File: MaxText/configs/base.yml (hunk 2 above)
Comment (gobbleturk) [reply MaxText/configs/base.yml]: Done.

File: MaxText/configs/base.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (gobbleturk) [reply MaxText/configs/base.yml:145]: Done.

**Pull request merged**